server/src/main/java/com/lunchtime/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,83 @@
+package com.lunchtime.service.impl;
+
+import com.lunchtime.models.Order;
+import com.lunchtime.models.OrderStatus;
+import com.lunchtime.models.Person;
+import com.lunchtime.models.RestaurantTable;
+import com.lunchtime.repository.OrderRepository;
+import com.lunchtime.service.OrderService;
+import com.lunchtime.service.OrderStatusService;
+import com.lunchtime.service.PersonService;
+import com.lunchtime.service.RestaurantTableService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+import java.util.Date;
+import java.util.List;
+import java.util.Optional;
+
+@RequiredArgsConstructor
+@Service
+public class OrderServiceImpl implements OrderService {
+
+    private final OrderRepository orderRepository;
+    private final RestaurantTableService restaurantTableService;
+    private final PersonService personService;
+    private final OrderStatusService orderStatusService;
+
+    private final String newOrderStatus = "new";
+
+    public Order saveOrder(Order order) {
+        Date currentDate = new Date();
+        if (currentDate.after(order.getStartTime())) {
+            return null;
+        }
+        if (order.getStartTime().after(order.getFinishTime())) {
+            return null;
+        }
+
+        List<Order> orders = orderRepository
+            .findAllOrdersByTableInTime(order.getTable().getId(), order.getStartTime(), order.getFinishTime());
+        if (!orders.isEmpty()) {
+            return null;
+        }
+
+        Optional<Person> person = personService.getPersonById(order.getPerson().getId());
+        if (person.isPresent()) {
+            order.setPerson(person.get());
+        } else {
+            return null;
+        }
+
+        Optional<RestaurantTable> table = restaurantTableService.getTableById(order.getTable().getId());
+        if (table.isPresent() && table.get().getCapacity() >= order.getVisitors()) {
+            order.setTable(table.get());
+        } else {
+            return null;
+        }
+
+        Optional<OrderStatus> status = orderStatusService.getOrderStatusByName(newOrderStatus);
+        status.ifPresent(order::setStatus);
+
+        return orderRepository.save(order);
```

Comment:
What app should do if status is not present?

Reply:
Nothing

Reply:
Due to impossibility of status absense, I find we can use simple `OrderStatus` here.

Reply:
I don't understand

Reply:
Do we can use smth like this?

```
OrderStatus status = orderStatusService.getOrderStatusByName(newOrderStatus);
order.setStatus(status);
```

Reply:
May be we need @ColumnDefault annotation in `OrderStatus` entity and nothing initialized here?
```
@NotNull
@Size(max = 32)
@Column(name = "name")
@ColumnDefault("new")
private String name;
```

Reply:
`getOrderStatusByName` returns `Optional`
It is hardcode logic, in next tasks we will add possibility to change order statuses
By default, seed file generates all statuses what we need

Reply:
As i know `@ColumnDefault("new")` is default value for any new order. I'm not sure if it is really hardcoded for this purpose. Also it can be changed any time

Reply:
Do you propose to remove OrderStatus entity, service, controller and seed function?

Reply:
I am surprised for your question. Why do you think I do that proposal for you, if I haven't?

Reply:
If you want to leave only "new" status, than other functionality aren't necessary, isn't?
`@ColumnDefault("new")` - is a bad idea, because of we can use "open" instead of "new", or other word

Reply:
1. Do we can change `@ColumnDefault("new")` value after it is persisted in DB?
2. As I understand, all orders created can have only one status until this order can be managed by restaurant (or user) ?